### PR TITLE
Pulse Weapons Sprite Update

### PR DESCRIPTION
**Pull request by Steelpoint**
The three main Pulse weapons now have sprites to show off which fire mode they are in. This affects all OBJECT sprites. However only the Rifle and Carbine show this in their MOB sprites, the Pistol still defaults to showing the standard gun MOB sprite.

pulse weapons
Top level Rifle, lower level Carbine. The mob sprite for the Carbine and Rifle is identical.

 :cl: Steelpoint
add: All Pulse weapons now accurately show, by their sprites, what fire mode they are in. 
/:cl:

### CODE REVIEW

**Comment by GunHog** [comment]
Awesome.

File: code/modules/projectiles/guns/energy/pulse.dm [hunk 1]
```diff
@@ -5,6 +5,7 @@
 	item_state = null
 	w_class = WEIGHT_CLASS_BULKY
 	force = 10
+	modifystate = 1
```
Comment (RemieRichards) [inline code/modules/projectiles/guns/energy/pulse.dm]: ```TRUE```

Comment (Steelpoint) [inline code/modules/projectiles/guns/energy/pulse.dm]: In code it states its either 0 or 1, this is how I figured out why the Pulse weapon sprites were not working when it came to swapping fire modes. modifystate = 1 is also how it works for energy_guns.

Comment (RemieRichards) [inline code/modules/projectiles/guns/energy/pulse.dm]: ```TRUE``` is ```1```, ```FALSE``` is ```0```
This is a boolean, treat it as such

Comment (coiax) [inline code/modules/projectiles/guns/energy/pulse.dm]: and TRUE is just a define for 1.

Comment (Steelpoint) [inline code/modules/projectiles/guns/energy/pulse.dm]: But that would be inconsistent with how every other weapon defines modifystate.

Comment (RemieRichards) [inline code/modules/projectiles/guns/energy/pulse.dm]: @coiax I did say that.
@Steelpoint the rest are wrong too!

Comment (Steelpoint) [inline code/modules/projectiles/guns/energy/pulse.dm]: I'll change mine around then.

Comment (Steelpoint) [inline code/modules/projectiles/guns/energy/pulse.dm]: true causes errors to pop up.

Comment (RemieRichards) [inline code/modules/projectiles/guns/energy/pulse.dm]: because it's ```TRUE```, not ```true```

**Comment by Steelpoint** [comment]
Updated for Boolean standard.

**Comment by WJohn** [comment]
Neat.

**Comment by adamkad1** [comment]
why would you ever switch off destroy?

**Comment by GunHog** [comment]
On the tiny chance that the murderboning antag you were sent to kill can be arrested and conscripted into the DeathSquad.

**Comment by Steelpoint** [comment]
More benefit for ERT who have more excuse to use stun over destroy.

Rare cases a DS might be there to not kill a particular person.

**Comment by Steelpoint** [comment]
Also, I hope to one day make unique MOB sprites for the pulse pistols. I think powerful weapons should have obvious mob sprites.

**Pull request merged**